### ISSUE TITLE
Support ChatCompletion streaming over Completions

### ISSUE THREAD

**Issue by aaronpowell** [microsoft/semantic-memory#96]
The OpenAI/AzureOpenAI `TextGeneration` classes use the legacy Completions API ([deprecation from OpenAI](https://platform.openai.com/docs/deprecations/2023-06-13-updated-chat-models, [announcement](https://openai.com/blog/gpt-4-api-general-availability) and [Azure model compatibility](https://learn.microsoft.com/azure/ai-services/openai/concepts/models#gpt-35-models)).

This poses a challenge when working with Azure OpenAI Service in particular as you are required to deploy a `gpt-35-turbo` model version of `0301`, which is deprecated, but due to quota limits on standard accounts you are unlikely to be able to deploy that plus the embeddings model and a `0613` model for `gpt-35-turbo` for the application to use.

**Comment by aaronpowell** [comment]
Ah, I just realised that I need to be using `AzureOpenAIConfig.APITypes.ChatCompletion` not `AzureOpenAIConfig.APITypes.TextCompletion` to go through the right codepath to use the new `ChatCompletions` endpoint, so this is probably not valid

**Comment by dluc** [comment]
correct, in the settings you can set the APIType to TextCompletion or ChatCompletion, depending on the model used
